src/providers/npm.py

```diff
@@ -0,0 +1,72 @@
+from typing import List, Tuple
+import requests
+import multiprocessing
+import itertools
+import semver
+import os
+
+from src.providers.Provider import Provider
+
+NPM_REGISTRY_URL = 'https://registry.npmjs.org/'
+
+
+def _get_version_package_payload(package_name: str, version: str) -> dict:
+    response = requests.get(NPM_REGISTRY_URL + package_name)
+    if response.status_code == 404:
+        raise Exception('Module not found {0}:{1}'.format(package_name, version))
+    if response.status_code > 399:
+        raise Exception('Unknown error occurred {0}:{1}'.format(package_name, version))
+    response_payload = response.json()
+    if version == 'latest':
```

Comment:
Both these `if`s seem unnecessary. won't `semver.max_satisfying` handle all cases?

Reply:
why would `node-semver` handle the http status code?

Reply:
anyway `max_satisfying(['1.1.0', '2.0.0'], 'latest', loose=False)` returns `None`, which is bad in our case because we want the `latest`

Reply:
fair enough. what about the 2nd `if`? e.g. `max_satisfying(['1.1.0', '2.0.0'], '1.1.0', loose=False)`

Reply:
`max_satisfying(['1.1.0', '2.0.0'], '1.1.0', loose=False) = '1.1.0'`

Reply:
2nd `if` was taken care of by removing it since by definition `latest` surely exists in `versions`